webui/src/components/versions.jsx

```diff
@@ -0,0 +1,94 @@
+import React from 'react/lib/ReactWithAddons';
+import { Link } from 'react-router'
+import { Map, List } from 'immutable';
+import { DateTimePicker, Multiselect, DropdownList, NumberPicker } from 'react-widgets';
+import moment from 'moment';
+import { serverCache, browser, Error } from '../data/server';
+import { keys, humanSize } from '../data/misc';
+import { ReplaceAnimate } from './animate.jsx';
+import { Wait, Err } from './waiting.jsx';
+
+
+const PT = React.PropTypes;
+
+
+export const Versions = React.createClass({
+    mixins: [React.addons.PureRenderMixin],
+
+    render() {
+        let {isDraft, recordID, versions} = this.props;
+        if (!versions) {
+            return false;
+        }
+        if (versions && versions.toJS) {
+            versions = versions.toJS();
+        }
+        const style = {
+            float:'right',
+            marginTop:'-3em',
+            marginBottom:'-3em',
+            color:'black',
+            padding:'15px',
+        };
+        return (
+            <div className="row">
+                <div className="col-sm-12" >
+                    { isDraft ?
+                        <DraftVersions draftID={recordID} versions={versions} style={style}/> :
+                        <PublishedVersions recordID={recordID} versions={versions} style={style}/> }
+                </div>
+            </div>
+        );
+    }
+});
+
+
+const DraftVersions = React.createClass({
+    mixins: [React.addons.PureRenderMixin],
+
+    render() {
+        let {draftID, versions, style} = this.props;
+        return (
+            <div style={style}>
+                You are now editing a draft of
+                <Link onClick={e=>browser.gotoRecord(versions[0].id)}> this published record</Link>.
+            </div>
+        );
+    },
+});
+
+const PublishedVersions = React.createClass({
```

Comment:
`return false` is sufficient here, we probably don't need to return an empty div.